system/classes/Kohana/Arr.php

```diff
@@ -101,7 +101,7 @@ public static function path($array, $path, $default = NULL, $delimiter = NULL)
 		}
 		else
 		{
-			if (array_key_exists($path, $array))
+			if (isset($path, $array))
```

Comment:
This is not going to work as expected. 

The easier solution here is not to pass an object into the `$array` value as this expected the value to be an array. Not an object

See: https://www.php.net/isset